### PR TITLE
Using verify_output() instead of expect_error()

### PR DESCRIPTION
**Pull request by romainfrancois**

### CODE REVIEW

File: R/join.r [hunk 1]
```diff
@@ -532,7 +532,7 @@ full_join.tbl_df <- function(x, y, by = NULL, copy = FALSE,
 
   out[by_x] <- vec_rbind(
     vec_slice(x[, by_x, drop = FALSE], x_indices_one),
-    vec_slice(y[, by_y, drop = FALSE], y_indices_two)
+    set_names(vec_slice(y[, by_y, drop = FALSE], y_indices_two), names(x)[by_x])
```
Comment (hadley) [inline R/join.r]: Can you please pull out into a separate PR?

File: tests/testthat/test-colwise-mutate-errors.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
+> with_lifecycle_errors({
```
Comment (hadley) [inline tests/testthat/test-colwise-mutate-errors.txt]: I think this is going to have merge conflicts with https://github.com/tidyverse/dplyr/pull/4706 😞

File: tests/testthat/test-colwise-filter.R [hunk 3]
```diff
@@ -78,3 +62,13 @@ test_that("colwise filter support .data$. in the quosure versions", {
     filter_at(iris, vars(contains(".")), any_vars(. > 4))
   )
 })
+
+
+# Errors ------------------------------------------------------------------
+
+test_that("colwise filter() give meaningful errors", {
```
Comment (hadley) [inline tests/testthat/test-colwise-filter.R]: For future reference, since we're planning to retire these functions, I wouldn't bother updating tests etc

File: tests/testthat/test-colwise-mutate-errors.txt (hunk 2 above)
Comment (hadley) [reply tests/testthat/test-colwise-mutate-errors.txt]: I think it might be easiest to merge my deprecations first, and then you can just revert any of the changes you've made to these files (since they don't need to be updated since the functions are deprecated)

File: R/grouped-df.r [hunk 4]
```diff
@@ -1,4 +1,4 @@
-utils::globalVariables(c("old_keys", "old_rows", ".rows", "new_indices", "new_rows", "new_rows_sizes", "needs_recycle"))
+utils::globalVariables(c("old_keys", "old_rows", ".rows", "new_indices", "new_rows", "new_rows_sizes", "needs_recycle", "distinct_vars"))
```
Comment (hadley) [inline R/grouped-df.r]: Can you put this in a separate PR too please?

File: R/join.r (hunk 1 above)
Comment (romainfrancois) [reply R/join.r]: https://github.com/tidyverse/dplyr/pull/4713

File: tests/testthat/test-colwise-filter.R (hunk 3 above)
Comment (romainfrancois) [reply tests/testthat/test-colwise-filter.R]: ah. I was just trying to universally use `verify_output()` everywhere. I can revert the changes. 

but is it not useful anyway to go through `verify_output()` as perhaps things that these functions use might change, giving slightly different error messages ... ?

Comment (hadley) [reply tests/testthat/test-colwise-filter.R]: Yeah, it's an improvement; I'm just not sure it was worth your time (but since you've already done it you should keep it)

Comment (romainfrancois) [reply tests/testthat/test-colwise-filter.R]: I'll follow through tomorrow once #4706 is live

**Comment by romainfrancois** [comment]
There are some new `expect_error()` calls in `test-slice.R` though

**Comment by romainfrancois** [comment]
... and some more rebasing to do 😬

**Comment by romainfrancois** [comment]
I think it's ok now, but then it might clash with #4715 ping @hadley

**Comment by hadley** [comment]
I think we should merge it and I'll deal with the aftermath

**Comment by hadley** [comment]
I'm going to merge now, so I can start updating my PR

**Comment by romainfrancois** [comment]
Thanks
